http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.java

```diff
@@ -212,7 +215,18 @@ private SdkEventLoopGroup nonManagedEventLoopGroup(SdkEventLoopGroup eventLoopGr
     @Override
     public void close() {
         runAndLogError(log, "Unable to close channel pools", pools::close);
-        runAndLogError(log, "Unable to shutdown event loop", sdkEventLoopGroup.eventLoopGroup()::shutdownGracefully);
+        runAndLogError(log, "Unable to shutdown event loop in 3000 milliseconds", () ->
+            closeEventLoopUninterruptibly(sdkEventLoopGroup.eventLoopGroup()));
+    }
+
+    private void closeEventLoopUninterruptibly(EventLoopGroup eventLoopGroup) throws
+                                                                              TimeoutException, ExecutionException {
+        try {
+            eventLoopGroup.shutdownGracefully().get(3000, TimeUnit.MILLISECONDS);
```

Comment:
shutdownGracefully has a timeout passed to it as well, that by default is longer than 3 seconds. Should we align these values in some way? Can we also catch errors we think can happen sometimes (eg. TimeoutException) to log a less scary error, if the customer doesn't need to worry about them?

Reply:
> Should we align these values in some way?

+1, it took me a while to understand the javadoc of `#shutdownGracefully(long, long, TimeUnit)`.  The default timeout is 15 seconds though and I will update it to 15 seconds.

https://github.com/netty/netty/blob/fa6a8cb09c9679468a6c2d912ddfbbe885ee0c08/common/src/main/java/io/netty/util/concurrent/AbstractEventExecutor.java#L37

> Can we also catch errors we think can happen sometimes (eg. TimeoutException) to log a less scary error, if the customer doesn't need to worry about them? 

Hmm, I do think customers need to be aware of the issue though as unclosed eventloop can cause resource leak. What do you think of changing the log level to warn?

Reply:
Oh, do you mean just catching `TimeoutException` and changing the error message? Yeah, we can do that. I'll update it